ignore_overflowの処理を見直し

sunflat opened the pull request:
ignore_overflow: true の条件を満たす時に、テキストが全部表示されない場合がある問題を解決するため

sunflat committed:
`valign: top` の時に、sectionを突き抜けない問題を修正

overflow: expand を取り除く処理を行わないと、sectionのboundsでトリミングされてしまうので、valign: top の時もこの処理を行う

sunflat committed:
section report からの描画でない時は、overflow: expand を取り除く処理をしないように

section report からの描画でない時は、overflow: expand を取り除くと、text枠を超えて描画できないようだ。（test:featuresの他のテストが失敗する）

sunflat commented:
![](https://github.trello.services/images/mini-trello-icon.png) [thinreports-generator に、overflow: expand に相当する機能を自前で実装する](https://trello.com/c/yvQStMPo/85-thinreports-generator-%E3%81%AB%E3%80%81overflow-expand-%E3%81%AB%E7%9B%B8%E5%BD%93%E3%81%99%E3%82%8B%E6%A9%9F%E8%83%BD%E3%82%92%E8%87%AA%E5%89%8D%E3%81%A7%E5%AE%9F%E8%A3%85%E3%81%99%E3%82%8B)

kosappi commented:
![](https://github.trello.services/images/mini-trello-icon.png) [thinreports-generator に、overflow: expand に相当する機能を自前で実装する→方針をみて欲しい](https://trello.com/c/yvQStMPo/85-thinreports-generator-%E3%81%AB%E3%80%81overflow-expand-%E3%81%AB%E7%9B%B8%E5%BD%93%E3%81%99%E3%82%8B%E6%A9%9F%E8%83%BD%E3%82%92%E8%87%AA%E5%89%8D%E3%81%A7%E5%AE%9F%E8%A3%85%E3%81%99%E3%82%8B%E2%86%92%E6%96%B9%E9%87%9D%E3%82%92%E3%81%BF%E3%81%A6%E6%AC%B2%E3%81%97%E3%81%84)

sunflat commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Thinreprots-generator: ignore-overflow の処理を見なおす](https://trello.com/c/zlLusbmX/157-thinreprots-generator-ignore-overflow-%E3%81%AE%E5%87%A6%E7%90%86%E3%82%92%E8%A6%8B%E3%81%AA%E3%81%8A%E3%81%99)

sunflat committed:
ignore_overflowの処理を見直し

tblock が follow-stretch: height かつ overflow: expand の時に、min bottom margin の影響で、section の auto-stretch が不十分だと、テキストが全部表示されない問題を解決するため

sunflat committed:
draw_shape_tblock に block を渡すパターンを、別メソッドにした

hidakatsuya commented:
This closes because this issue will be solved in another solution.
